shared/src/main/scala/bloop/io/Environment.scala

```diff
@@ -0,0 +1,174 @@
+   *    '\n'   everywhere else
+   *
+   * The default ending in Windows is '\r\n', but either of two conditions can
+   * change the value of [[shellLineSeparator]]:
+   *    1. if [[validShell]] is nonEmpty
+   *    2. if system property `"line.separator"` is modified
+   *
+   * Unlike [[lineSeparator]], only depends on the value of SHELL.
+   *
+   * @return preferred lineSeparator for SHELL environments, or `System.lineSeparator` otherwise.
+   */
+  def shellLineSeparator: String = if (validShell.nonEmpty) "\n" else sysLineSeparator
+
+  /**
+   * Preferred line ending for SHELL environments.
+   *
+   * An alias for `shellLineSeparator`.
+   *
+   * @example
+   * {{{
+   *   print(someText + EOL)
+   * }}}
+   */
+  lazy val EOL: String = shellLineSeparator // alias for shellLineSeparator
+
+  /**
+   * Return the canonical lowercased shell name, or empty string.
+   *
+   * `shell` is valid if when lowercased, it has a `validShells` entry as substring.
+   *
+   * @return the matching `validShells` entry, if there is a match; empty string otherwise.
+   *
+   * @example if {{{SHELL=/usr/bin/bash}}}, returns {{{/bin/bash}}}.
+   *
+   */
+  def validShell: String = validShells.find { _.endsWith(shellLC) } match {
+    case Some(sh) => sh
+    case _ => ""
+  }
+
+  /*
+   * Return lowercase shell with possible .exe extension removed.
+   */
+  private[this] def shellLC = shell.toLowerCase(java.util.Locale.ENGLISH) match {
+    case sh if sh.endsWith(".exe") => sh.replaceFirst(".exe$", "")
+    case sh => sh
+  }
+
+  /*
+   * Return the name of the SHELL environment variable, or empty string.
+   *
+   * Normally set in CYGWIN, MinGW, msys, Git Bash shell environments.
+   */
+  lazy val shell: String = getenvOpt("SHELL").getOrElse("")
+
+  /*
+   * Returns current value of system property `"line.separator"`.
+   *
+   * System.lineSeparator may only be redefined at JVM startup,
+   * whereas this may be redefined on the fly in client code.
+   *
+   * System.lineSeparator always returns the initial value of system property "line.separator".
+   * @see [[https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#lineSeparator--]]
+   *
+   * System properties may be redefined on JVM startup, e.g., from a shell environment:
+   * @example
+   *  {{{
+   *  $ java -J-Dline.separator=$'\n' ...
+   *  }}}
+   *
+   */
+  def sysLineSeparator: String = propsOpt("line.separator").getOrElse("\n")
+
+  def propsOpt(propname: String): Option[String] = Option(sys.props(propname))
+
+  def getenvOpt(varname: String): Option[String] = Option(System.getenv(varname))
+
+  /*
+   * The safe general way to split Strings to lines.   Makes minimal assumptions
+   * about the source of the String.
+   *
+   * Does not depend on editor end-of-line configuration.
+   * Correctly splits strings from any of the common OSTYPES.
+   */
+  def lineSplit(str: String): Array[String] = str.split(END_OF_LINE_MATCHER, -1)
```

Comment:
I think this one is only used in one place, we can remove it and just use `LineSplitter`